lib/core/Canvas.js

```diff
@@ -1153,7 +1153,7 @@ Canvas.prototype._viewboxChanged = function() {
 Canvas.prototype.viewbox = function(box) {
 
   if (box === undefined && this._cachedViewbox) {
-    return this._cachedViewbox;
+    return JSON.parse(JSON.stringify(this._cachedViewbox));
```

Comment:
The problem is when you return a reference to the viewbox to the caller of `viewbox()` they will assume that it's not going to be mutated. This is an issue I ran into. I called `viewbox()` to then used the viewbox at a later point just to realize it had been changed.

Reply:
BTW there is a dedicated API for deep cloning: https://developer.mozilla.org/en-US/docs/Web/API/Window/structuredClone

Reply:
Could be a follow-up improvement.

Reply:
Simple clone: 

```
{ ...viewbox }
```

Reply:
I'll need to figure why this needs a copy in the first place. [We always construct a fresh viewbox.](https://github.com/bpmn-io/diagram-js/blob/develop/lib/core/Canvas.js#L1184), and the cached viewbox is not to be fiddled with.

@philippfromme Still looking for a concrete scenario where users would mutate the viewbox, and would love to see that as a test.

To prevent mutation (which does not make sense anyway), we can [freeze the object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze), too.